.github/workflows/main.yml

```diff
@@ -0,0 +1,35 @@
+name: CI
+
+on: [push, pull_request] # on all pushes and PRs
+
+jobs:
+  ros1_bridge:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        distro: ["ros:rolling-ros1-bridge-focal", "ros:foxy-ros1-bridge-focal"]
+    container:
+      image: ${{ matrix.distro }}
+    env:
+      CCACHE_DIR: "/home/runner/.ccache"
+    steps:
+      - uses: actions/checkout@v2
+      - name: install ccache
+        run: |
+          sudo apt-get update -y
+          sudo apt-get -qq -y install ccache
+      - name: ccache cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ccache-${{ matrix.distro }}-${{github.run_id}}
+          restore-keys: |
+            ccache-${{ matrix.distro }}
+      - name: ccache stats
+        run: ccache -s
+      - name: Build ros1-bridge
```

Comment:
Have you considered the ros-tooling GitHub actions, [setup-ros](https://github.com/ros-tooling/setup-ros) and [action-ros-ci](https://github.com/ros-tooling/action-ros-ci)? I think it would be nice to standardize on these tools compared to possibly duplicating work/maintenance.

Reply:
Do these actions support mixed systems? (ROS1 and ROS2)? The reason why I did not use [industrial_ci](https://github.com/ros-industrial/industrial_ci) is because it currently does not support mixed systems.

Reply:
I haven't tried mixing ROS 1 and ROS 2, but it sounds like it is supported based on the action-ros-ci README. We should be able to do something like:

```
  - uses: ros-tooling/action-ros-ci@v0.2
    with:
      package-name: ros1_bridge
      target-ros1-distro: noetic
      target-ros2-distro: foxy
```

Reply:
This is the config now
```
      - uses: ros-tooling/setup-ros@v0.1
        with:
          required-ros-distributions: "noetic rolling"
      - name: Build and test ros1-bridge
        uses: ros-tooling/action-ros-ci@v0.2
        with:
          package-name: ros1_bridge
          target-ros1-distro: noetic
          target-ros2-distro: rolling
```
[flake tests](https://github.com/ipa-hsd/ros1_bridge/runs/2446964399?check_suite_focus=true#step:4:1218) fail, but [actions/upload-artifact](https://github.com/actions/upload-artifact) is not executed since the previous step failed. I cannot see the log

Reply:
Cool, looks promising! I can see the log; there's a couple "blind except" warnings:

```
2021-04-27T10:37:43.5835983Z [4.810s] test 4
2021-04-27T10:37:43.5836342Z [4.810s]       Start  4: flake8
2021-04-27T10:37:43.5836673Z [4.810s] 
2021-04-27T10:37:43.5927752Z [4.810s] 4: Test command: /usr/bin/python3 "-u" "/opt/ros/rolling/share/ament_cmake_test/cmake/run_test.py" "/home/runner/work/ros1_bridge/ros1_bridge/ros_ws/build/ros1_bridge/test_results/ros1_bridge/flake8.xunit.xml" "--package-name" "ros1_bridge" "--output-file" "/home/runner/work/ros1_bridge/ros1_bridge/ros_ws/build/ros1_bridge/ament_flake8/flake8.txt" "--command" "/opt/ros/rolling/bin/ament_flake8" "--xunit-file" "/home/runner/work/ros1_bridge/ros1_bridge/ros_ws/build/ros1_bridge/test_results/ros1_bridge/flake8.xunit.xml"
2021-04-27T10:37:43.5929928Z [4.810s] 4: Test timeout computed to be: 60
2021-04-27T10:37:43.5931216Z [4.863s] 4: -- run_test.py: invoking following command in '/home/runner/work/ros1_bridge/ros1_bridge/ros_ws/src/3pniospi2vh/ros1_bridge':
2021-04-27T10:37:43.5932687Z [4.863s] 4:  - /opt/ros/rolling/bin/ament_flake8 --xunit-file /home/runner/work/ros1_bridge/ros1_bridge/ros_ws/build/ros1_bridge/test_results/ros1_bridge/flake8.xunit.xml
2021-04-27T10:37:43.5933496Z [5.624s] 4: 
2021-04-27T10:37:43.5934011Z [5.624s] 4: ./ros1_bridge/__init__.py:291:1: B902 blind except Exception: statement
2021-04-27T10:37:43.5934640Z [5.624s] 4:                     except Exception as e:
2021-04-27T10:37:43.5935032Z [5.624s] 4: ^
2021-04-27T10:37:43.5935344Z [5.624s] 4: 
2021-04-27T10:37:43.5935866Z [5.624s] 4: ./ros1_bridge/__init__.py:344:1: B902 blind except Exception: statement
2021-04-27T10:37:43.5936474Z [5.624s] 4:                     except Exception as e:
2021-04-27T10:37:43.5936882Z [5.625s] 4: ^
2021-04-27T10:37:43.5937280Z [5.625s] 4: 
2021-04-27T10:37:43.5937733Z [5.625s] 4: 2     B902 blind except Exception: statement
2021-04-27T10:37:43.5938164Z [5.625s] 4: 
2021-04-27T10:37:43.5938511Z [5.625s] 4: 1 files checked
2021-04-27T10:37:43.5938871Z [5.625s] 4: 2 errors
```

I think we can just quiet those warnings by appending `# noqa: B902` to the reported lines.

Reply:
And we have a [winner](https://github.com/ipa-hsd/ros1_bridge/runs/2455276524?check_suite_focus=true#step:4:1253)! I will update the config for `ccache`.